xdp-dump/xdpdump.c

```diff
@@ -709,16 +852,73 @@ static bool capture_on_interface(struct dumpopt *cfg)
 
         /* Open the pcap handle */
 	if (cfg->pcap_file) {
-		pcap = pcap_open_dead(DLT_EN10MB, cfg->snaplen);
-		if (!pcap) {
-			pr_warn("ERROR: Can't open pcap dead handler!\n");
-			goto error_exit;
-		}
 
-		pcap_dumper = pcap_dump_open(pcap, cfg->pcap_file);
-		if (!pcap_dumper) {
-			pr_warn("ERROR: Can't open pcap file for writing!\n");
-			goto error_exit;
+		if (cfg->use_pcap) {
+			pcap = pcap_open_dead(DLT_EN10MB, cfg->snaplen);
+			if (!pcap) {
+				pr_warn("ERROR: Can't open pcap dead handler!\n");
+				goto error_exit;
+			}
+
+			pcap_dumper = pcap_dump_open(pcap, cfg->pcap_file);
+			if (!pcap_dumper) {
+				pr_warn("ERROR: Can't open pcap file for writing!\n");
+				goto error_exit;
+			}
+		} else {
+			char           if_name[IFNAMSIZ + 7];
+			char           if_descr[BPF_OBJ_NAME_LEN + IFNAMSIZ + 10];
+			char           if_drv[260];
```

Comment:
I'm not doubting that you calculated the size of these buffers correctly, but it does make me a bit twitchy to see these + no checks of snprintf() return codes. At least use try_snprintf(), as I pointed out above :)

Reply:
Will fix all snprintf() calls in the code